### DISCUSSION TITLE
GPU used for benchmarking

### DISCUSSION THREAD

**Discussion by CyrusAddy** [ultralytics/ultralytics#21822] [Q&A]
Can you help me with the GPU specifications employed for the benchmarking of the YOLO algorithms presented in this image?

<img width="900" height="385" alt="image" src="https://github.com/user-attachments/assets/899376da-f828-4e02-bdbb-f8d0b9a91096" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @CyrusAddy, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will follow up soon to assist further.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To help us clarify your benchmarking question, please share:
- The exact image/figure source or link within the repository/docs
- The specific models of interest (e.g., YOLO11 variants)
- Your current environment details (OS, Python, PyTorch, CUDA) and how you plan to reproduce the benchmarks

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Thanks for the question, @CyrusAddy. Could you share the exact docs/repo link for that figure so I can confirm the precise GPU used? We publish multiple benchmark charts across devices. For the Jetson charts specifically, the hardware is Jetson AGX Orin 64GB, Jetson Orin NX 16GB, and Jetson Orin Nano Super, evaluated at FP32 and imgsz=640; see the NVIDIA Jetson guide for details: [NVIDIA Jetson with Ultralytics YOLO11](https://docs.ultralytics.com/guides/nvidia-jetson/). To reproduce desktop benchmarks on your own GPU and log device specs, you can run:

```python
from ultralytics.utils.benchmarks import benchmark
benchmark(model="yolo11n.pt", data="coco8.yaml", imgsz=640, device=0)
```

More on benchmarking is here: [Benchmark mode](https://docs.ultralytics.com/modes/benchmark/) and the API reference: [benchmark() reference](https://docs.ultralytics.com/reference/utils/benchmarks/#ultralytics.utils.benchmarks.benchmark).

**Reply by CyrusAddy** [reply]
@glenn-jocher This is the link to the actual image https://www.themoonlight.io/en/review/comparing-yolov11-and-yolov8-for-instance-segmentation-of-occluded-and-non-occluded-immature-green-fruits-in-complex-orchard-environment

Name of paper: COMPARING YOLOV11 AND YOLOV8 FOR INSTANCE
SEGMENTATION OF OCCLUDED AND NON-OCCLUDED IMMATURE
GREEN FRUITS IN COMPLEX ORCHARD ENVIRONMENT

**Reply by CyrusAddy** [reply]
@glenn-jocher Is it possible to perform a benchmark on my custom data from scratch on my local PC or Kubernetes? If possible, please provide a CLI for that. Thanks.

**Reply by glenn-jocher** [reply]
Thanks for the link—since that figure is from a third‑party study, we can’t confirm the GPU they used; please check the paper’s methods/hardware section or contact the authors.

Yes, you can benchmark on your custom data locally or in Kubernetes (run the same CLI inside your container):
```bash
# Detection example
yolo benchmark model=path/to/model.pt data=path/to/data.yaml imgsz=640 device=0

# Segmentation example
yolo benchmark model=yolo11n-seg.pt data=path/to/seg_dataset.yaml imgsz=640 device=0

# Optional: benchmark a specific export format (e.g., ONNX)
yolo benchmark model=path/to/model.pt data=path/to/data.yaml imgsz=640 device=0 format=onnx
```
Ensure your dataset YAML matches the task and paths are accessible; see the [Segmentation task docs](https://docs.ultralytics.com/tasks/segment/) for dataset format details.

**Reply by CyrusAddy** [reply]
@glenn-jocher Is that to say you can't benchmark using the yaml file of the model but only the best,pt of the COCO pre-trained model? Is that the case?

**Reply by glenn-jocher** [reply]
Not the case—`yolo benchmark` works with any trained `.pt` weights, including your own `runs/.../weights/best.pt`; use your weights in `model=...` and your dataset YAML in `data=...`. A model YAML alone builds an untrained network, so accuracy metrics won’t be meaningful. See the [Train mode docs](https://docs.ultralytics.com/modes/train/) for producing `best.pt`.

**Comment by Y-T-G** [comment]
The plot in the image was benchmarked on NVIDIA T4.
